alacritty_terminal/src/renderer/rects.rs

```diff
@@ -35,69 +35,66 @@ impl<T> Rect<T> {
 }
 
 struct Line {
-    rect: Rect<f32>,
     start: Point,
     end: Point,
     color: Rgb,
 }
 
-impl Line {
-    /// Create a line that starts on the left of `cell` and is one cell wide
-    fn from_cell(cell: &RenderableCell, flag: Flags, metrics: &Metrics, size: &SizeInfo) -> Line {
-        let cell_x = cell.column.0 as f32 * size.cell_width;
-
-        let (position, mut height) = match flag {
-            Flags::UNDERLINE => (metrics.underline_position, metrics.underline_thickness),
-            Flags::STRIKEOUT => (metrics.strikeout_position, metrics.strikeout_thickness),
-            _ => unimplemented!("Invalid flag for cell line drawing specified"),
-        };
-
-        // Make sure lines are always visible
-        height = height.max(1.);
-
-        let cell_bottom = (cell.line.0 as f32 + 1.) * size.cell_height;
-        let baseline = cell_bottom + metrics.descent;
-
-        let mut y = baseline - position - height / 2.;
-        let max_y = cell_bottom - height;
-        if y > max_y {
-            y = max_y;
-        }
-
-        let rect = Rect::new(cell_x + size.padding_x, y + size.padding_y, size.cell_width, height);
-
-        Self { start: cell.into(), end: cell.into(), color: cell.fg, rect }
-    }
+/// Lines for underline and strikeout.
+#[derive(Default)]
+pub struct Lines {
+    inner: HashMap<Flags, Vec<Line>>,
+}
 
-    fn update_end(&mut self, end: Point, size: &SizeInfo) {
-        self.rect.width = (end.col + 1 - self.start.col).0 as f32 * size.cell_width;
-        self.end.col = end.col;
+fn create_rect(
```

Comment:
This should probably be moved into `Line`, since it's exclusive to that. It even takes in a `&Line`, so that could just be `&self`.

Reply:
Kind of, just placed it here for now, since wasn't sure.

Reply:
Biggest issue here is probably the naming. I think `impl From<Line> for Rect` might do the job?

Reply:
>  I think impl From<Line> for Rect might do the job?

That seems better.

Reply:
Unfortunately that shouldn't work, since there is a restriction of only 1 parameter in from. However we can add custom method `from_line` to `Rect`.

Reply:
I think `from_line` makes sense on `Rect`, yeah.